src/Symfony/Component/Notifier/Bridge/Firebase/FirebaseOptions.php

```diff
@@ -27,10 +27,13 @@ abstract class FirebaseOptions implements MessageOptionsInterface
      */
     protected $options;
 
-    public function __construct(string $to, array $options)
+    protected $data;
+
+    public function __construct(string $to, array $options, array $data)
```

Comment:
I think I would prefer to use a key `data` inside options array, WDYT?

Reply:
Yes, I get what you're saying. The con for this is the `toArray` method that returns everything that is in `options` array for key 'notification' and I would have to do some logic either in `FirebaseTransport` or here in `toArray` to retrieve what's there under the `data` key. 
But it is doable indeed. :smile:

Reply:
Maybe it makes sense, but I don't get it 😄 
When you add it to the options array and return it in `toArray`, we can use it there and if the (I hope I understood this right) transport is instantiated without `FirebaseOptions` we add the 'data' key to the payload as an empty array.

The main point is: Whenever I send a payload it needs to include a `data` key, right?

Reply:
the `options` in the constructor does not actually represent the full options. It represents only the `notication` key in the firebase options (see `toArray`. To me, that's a naming issue in the `FirebaseOptions` class.

Reply:
I get what you are saying @OskarStark , but from my point of view, that `options` field is indeed a naming issue there and I suggest, for the sake of clarity in mapping the firebase api, to change it to something like `notificationOptions` and add a separate `data` property.

Reply:
@stof do you think I should rename that field? I would really need this PR to be merged soon because I am a bit blocked on my implementation for push notifications on the project I'm currently working on.
Thanks :+1: